src/test/lib/scheduler/scheduler_test.cpp

```diff
@@ -393,4 +393,26 @@ TEST_F(SchedulerTest, NodeQueueSchedulerCreationAndReset) {
   }
 }
 
+TEST_F(SchedulerTest, ShutdownTaskDecrement) {
+  // Inherit to overwrite execute which would otherwise force us to execute the tasks by scheduling them.
+  struct ShutdownTaskTest : public ShutdownTask {
+    explicit ShutdownTaskTest(std::atomic_int64_t& active_worker_count) : ShutdownTask{active_worker_count} {}
+
+    void execute() {
+      _on_execute();
+    }
+  };
```

Comment:
`AbstractTask::execute()` is public,right? Or do I miss an `Assert()` somewhere?

Reply:
Yes. Public. Not sure I understand what you are meaning though.

Reply:
Wondered why we need the wrapper class. It's this `Assert()`, right?
https://github.com/hyrise/hyrise/blob/6681524055b68b3225eb29125fd350bc7d945cff/src/lib/scheduler/abstract_task.cpp#L230

Getting the task in `TaskState::AssignedToWorker` should also be possible with this public function:
https://github.com/hyrise/hyrise/blob/6681524055b68b3225eb29125fd350bc7d945cff/src/lib/scheduler/abstract_task.cpp#L88-L90

Thus, the following should work w/o the wrapper class:
```c++
auto task = ShutdDownTask{counter};
task.try_mark_as_assigned_to_worker();
task.execute();
```

Reply:
Hmm, let me think about that. Not sure yet if I like it.